### PR TITLE
Azure Traffic Manager implementation with multiple aks clusters

### LINKED ISSUE #42: Deploy multiple clusters with ingress and Azure Traffic Manager
Add scripts to enable deploying multiple clusters and have them managed by a common ingress (traffic manager).

### PR DESCRIPTION
**Pull request by sarath-p**
* Implements Azure traffic manager profile module in a global Azure resource group
* Implements Azure traffic manager endpoint and public ip module
* Implements two aks clusters in two different azure regions
* Supports aks cluster load balancer public ip configuration with Azure traffic manager end point.
* Adds role assignment with Contributor role for aks service principal so it's loadbalancer joins vnet/subnet which are in non aks node resource group.
* Adding location-clustername prefix to the kube config file name to avoid the overwrite.
* Cloning and installing Flux from unique directory for each cluster and location.
* Updated kube config copy instructions in readme to include <location-cluster> prefix in the file name.
* Removed bedrock implementation of Azure DevOps git repo implementation since flux supports by default.

Closes #42

### CODE REVIEW

File: cluster/azure/tm-endpoint-ip/variables.tf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+variable "ip_address_filename" {
+  type    = "string"
+  default = "bedrock_public_ip_address"
+}
+
+variable "ipaddress_to_disk" {
+  description = "This disables or enables the ip address output file from being written to disk."
+  type        = "string"
+  default     = "true"
```
Comment (dtzar) [inline cluster/azure/tm-endpoint-ip/variables.tf]: My vote is to set this to false by default so we don't have the linux requirement & processing time for it. However, this information is useful, so we should include the output for the public IP so people can still input to their [Kubernetes yaml](https://docs.microsoft.com/en-us/azure/aks/static-ip#create-a-service-using-the-static-ip-address).

File: cluster/azure/tm-profile/variables.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+variable "resource_group_name" {
+  type = "string"
+}
+
+variable "resource_group_location"{
+  type="string"
+}
+
+# variable "public_ips" {
```
[the remark this answers is not in the document]
Comment (sarath-p) [reply cluster/azure/tm-profile/variables.tf]: it was unused and commented, but removing now.

File: cluster/azure/vnet/variables.tf [hunk 3]
```diff
@@ -8,6 +8,11 @@ variable "resource_group_name" {
   default     = "myapp-rg"
 }
 
+variable "resource_group_location" {
+  description = "Default resource group location that the resource group will be created in. The full list of Azure regions can be found at https://azure.microsoft.com/regions"
+  default     = "westus2"
```
[the remark this answers is not in the document]
Comment (sarath-p) [reply cluster/azure/vnet/variables.tf]: Ok, will remove the default so becomes required.

File: cluster/azure/tm-endpoint-ip/variables.tf (hunk 1 above)
Comment (sarath-p) [reply cluster/azure/tm-endpoint-ip/variables.tf]: that makes sense and I agree that it makes finding ip address easier for configuration of load balancer yaml.

File: cluster/azure/tm-profile/main.tf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  resource_group_name    = "${azurerm_resource_group.tmrg.name}"
+  traffic_routing_method = "Weighted"
+
+  dns_config {
+    relative_name = "${var.traffic_manager_dns_name}"
+    ttl           = 30
+  }
+
+  monitor_config {
+    protocol = "http"
```
[the remark this answers is not in the document]
Comment (sarath-p) [reply cluster/azure/tm-profile/main.tf]: should we make port configurable as well?

[the remark this answers is not in the document]
Comment (dtzar) [reply cluster/azure/tm-profile/main.tf]: yes

File: cluster/common/flux/deploy_flux.sh [hunk 5]
```diff
@@ -1,27 +1,35 @@
...
-REPO_DIR="flux"
+REPO_DIR="$REPO_ROOT_DIR/flux"
 FLUX_CHART_DIR="flux/chart/flux"
 FLUX_MANIFESTS="manifests"
 
-GIT_KNOWN_HOSTS="`ssh-keyscan github.com gitlab.com bitbucket.org ssh.dev.azure.com`"
+echo "flux repo root directory: $REPO_ROOT_DIR"
 
-echo "known_hosts $GIT_KNOWN_HOSTS"
+rm -rf $REPO_ROOT_DIR
```
[the remark this answers is not in the document]
Comment (sarath-p) [reply cluster/common/flux/deploy_flux.sh]: It was a unused variable, but cleaned it up.

File: cluster/common/flux/deploy_flux.sh [hunk 6]
```diff
@@ -58,7 +67,7 @@ echo "creating kubernetes secret $KUBE_SECRET_NAME from key file path $GITOPS_SS
 kubectl create secret generic $KUBE_SECRET_NAME --from-file=identity=$GITOPS_SSH_KEY -n $KUBE_NAMESPACE
 
 echo "Applying flux deployment"
-if ! kubectl apply -f  $FLUX_CHART_DIR/$FLUX_MANIFESTS/flux/templates -n $KUBE_NAMESPACE; then
+if ! kubectl apply -f  $REPO_ROOT_DIR/$FLUX_CHART_DIR/$FLUX_MANIFESTS/flux/templates -n $KUBE_NAMESPACE; then
```
[the remark this answers is not in the document]
Comment (sarath-p) [reply cluster/common/flux/deploy_flux.sh]: It was a unused variable, but cleaned it up.

**Pull request merged**